src/config/database_pools.rs

```diff
@@ -20,15 +24,30 @@ pub struct DatabasePools {
 pub struct DbPoolConfig {
     pub url: String,
     pub read_only_mode: bool,
+    pub pool_size: u32,
+    pub min_idle: Option<u32>,
 }
 
 impl DatabasePools {
     pub fn are_all_read_only(&self) -> bool {
         self.primary.read_only_mode
     }
+
+    pub fn primary_max_pool_size(&self) -> u32 {
+        self.primary.pool_size
+    }
+
+    pub fn replica_max_pool_size(&self) -> u32 {
+        self.replica
+            .as_ref()
+            .map(|config| config.pool_size)
+            .unwrap_or(Self::DEFAULT_POOL_SIZE)
+    }
```

Comment:
This function doesn't seem to be used anywhere.

Reply:
Good catch!